InnerEye/ML/run_ml.py

```diff
@@ -669,3 +676,29 @@ def run_model_test(data_split: ModelExecutionMode) -> Optional[InferenceMetrics]
                         train_metrics=train_metrics, run_context=run_context)  # type: ignore
 
         return test_metrics, val_metrics, train_metrics
+
+    def download_weights(self) -> Path:
+        target_folder = self.project_root / fixed_paths.MODEL_WEIGHTS_DIR_NAME
+        target_folder.mkdir(exist_ok=True)
+
+        url = self.model_config.weights_url
+
+        # assign the same filename as in the download url if possible, so that we can check for duplicates
+        # If that fails, map to a random uuid
+        file_name = os.path.basename(requests.utils.urlparse(url).path) or str(uuid.uuid4().hex)
+        result_file = target_folder / file_name
+
+        # only download if hasn't already been downloaded
+        if result_file.exists():
+            logging.info(f"File already exists, skipping download: {result_file}")
+            return result_file
+
+        logging.info(f"Downloading weights.")
+
+        response = requests.get(url, stream=True)
+        response.raise_for_status()
+        with open(result_file, "wb") as file:
+            for chunk in response.iter_content(chunk_size=1024):
+                file.write(chunk)
```

Comment:
is there not library function that does that? It's such a common use case, I would have thought?

Reply:
As far as I can find, there is nothing in the requests library that handles writing directly to file.